flight/Modules/Logging/logging.c

```diff
@@ -602,42 +627,47 @@ static void register_default_profile()
  */
 static void writeHeader()
 {
+	uint8_t fw_blob[100];
+	PIOS_BL_HELPER_FLASH_Read_Description((uint8_t *) &fw_blob,
+			sizeof(fw_blob));
+
 	int pos;
 #define STR_BUF_LEN 45
 	char tmp_str[STR_BUF_LEN];
 	char *info_str;
 	char this_char;
 	DateTimeT date_time;
 
-	const struct pios_board_info * bdinfo = &pios_board_info_blob;
-
 	// Header
 	#define LOG_HEADER "dRonin git hash:\n"
 	send_data((uint8_t *)LOG_HEADER, strlen(LOG_HEADER));
 
 	// Commit tag name
-	info_str = (char*)(bdinfo->fw_base + bdinfo->fw_size + 14);
-	send_data((uint8_t*)info_str, strlen(info_str));
+	// XXX all of thse should use the fw_version_info structure instead of
+	// doing offset math, but we need to factor out the fw_version_info
+	// struct from the templates first.
+	info_str = (char*)(fw_blob + 14);
+	send_data((uint8_t*)info_str, strnlen(info_str, 26));
```

Comment:
Good call on using `strnlen` it was crashing for me the other day when `fw_blob` was all `FF`. Was on my list of things to fix :).